rest_api/test/test_rest_api.py

```diff
@@ -5,25 +5,33 @@
 from textwrap import dedent
 from unittest import mock
 from unittest.mock import MagicMock, Mock
-import functools
 import numpy as np
 import pandas as pd
 
 import pytest
 from fastapi.testclient import TestClient
-from haystack import Document, Answer
+from haystack import Document, Answer, Pipeline
 import haystack
 from haystack.nodes import BaseReader, BaseRetriever
 from haystack.document_stores import BaseDocumentStore
 from haystack.schema import Label, FilterType
 from haystack.nodes.file_converter import BaseConverter
 
+from rest_api.pipeline import _setup_indexing_pipeline
 from rest_api.utils import get_app
 
 
 TEST_QUERY = "Who made the PDF specification?"
 
 
+def test_check_single_worker_warning_for_indexing_pipelines(caplog):
+    yaml_pipeline_path = Path(__file__).parent.resolve() / "samples" / "test.in-memory-haystack-pipeline.yml"
+    p = _setup_indexing_pipeline(yaml_pipeline_path, None)
+
+    assert isinstance(p, Pipeline)
+    assert "used with 1 worker" in caplog.text
+
+
```

Comment:
We can add another case to test the try/except PipelineConfigError by trying to load a bogus pipeline and assert the function returns `None`

Reply:
Added